src/index.ts

```diff
@@ -215,6 +215,9 @@ module.exports = class ServerlessPlugin {
       } catch (err) {
         if (err instanceof Error) {
           this.serverless.cli.log(`Error occurred when configuring monitors: ${err.message}`);
+          if (config.failOnError) {
+            throw err;
```

Comment:
Should we log something? Or does the throw on line 219 get caught by line 216?

Reply:
Great question! Since this plugin is for the Serverless framework, we're using `this.serverless.cli.log` to print the message to the command line.

Previously we'd consider this a non fatal error and continue, this PR changes this behavior to make the error fatal (and thus fail CI like github actions, jenkins, gitlab, etc) - only if the configuration option is set.